pkg/cmd/cli/describe/projectstatus.go

```diff
@@ -35,59 +37,41 @@ type ProjectStatusDescriber struct {
 	C client.Interface
 }
 
+type GraphLoadingFunc func(g osgraph.Graph, graphLock sync.Mutex, namespace string, kclient kclient.Interface, client client.Interface) error
+
 func (d *ProjectStatusDescriber) MakeGraph(namespace string) (osgraph.Graph, error) {
 	g := osgraph.New()
 
-	svcs, err := d.K.Services(namespace).List(labels.Everything())
-	if err != nil {
-		return g, err
-	}
-
-	iss, err := d.C.ImageStreams(namespace).List(labels.Everything(), fields.Everything())
-	if err != nil {
-		return g, err
-	}
-
-	bcs, err := d.C.BuildConfigs(namespace).List(labels.Everything(), fields.Everything())
-	if err != nil {
-		return g, err
-	}
-
-	dcs, err := d.C.DeploymentConfigs(namespace).List(labels.Everything(), fields.Everything())
-	if err != nil {
-		return g, err
-	}
-
-	builds := &buildapi.BuildList{}
-	if len(bcs.Items) > 0 {
-		if b, err := d.C.Builds(namespace).List(labels.Everything(), fields.Everything()); err == nil {
-			builds = b
+	listingWaitGroup := sync.WaitGroup{}
+	graphLock := sync.Mutex{}
+	errorChannel := make(chan error, 100)
```

Comment:
> why 100?

Big enough that no reasonable number of resource gets will block writers since the consumption of the channel happens after the wait so that I can close it before reading it.  I thought that factorization was easier to read than a clever `select`, but I'm willing to change it.